com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java

```diff
@@ -310,41 +321,53 @@ public static TextureSetResult calculateLayout(List<Rect> images, List<SpriteGeo
                 .collect(Collectors.toList());
 
         List<Layout> layouts;
-        List<Rect> layoutRects;
-        int layoutWidth;
-        int layoutHeight;
```

Comment:
This bit is fine, but I noticed we mutate the input `images` `Rects` in this bit above:
```java
// Store rectangle order as the AnimIterator interface relies on stable frame indices.
for(int i = 0; i < images.size(); i++) {
     images.get(i).index = i;
}
```
It would be cool it you left the input alone and instead assigned the `index` in the `resizedImages` stream expression, as we cannot cache anything that is mutated.

The Bob code plays kind of fast and loose with in-place mutation, but I've looked over the remaining code that's being called and I'm pretty sure we should be fine apart from this.